src/Symfony/Component/Security/Http/EventListener/IsCsrfTokenValidAttributeListener.php

```diff
@@ -37,9 +41,12 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
         }
 
         $request = $event->getRequest();
+        $arguments = $event->getNamedArguments();
 
         foreach ($attributes as $attribute) {
-            if (!$this->csrfTokenManager->isTokenValid(new CsrfToken($attribute->id, $request->request->getString($attribute->tokenKey)))) {
+            $id = $this->getIsCsrfTokenValidId($attribute->id, $request, $arguments);
+
+            if (!$this->csrfTokenManager->isTokenValid(new CsrfToken($id, $request->request->getString($attribute->tokenKey)))) {
```

Comment:
CsrfToken() $id argument must be a string.

Reply:
updated with a proposition, any other proposition is welcomed

Reply:
made `getIsCsrfTokenValidId` return type a string